### PR TITLE
Registration endpoint

### PR DESCRIPTION
**Pull request by Darjusch**
Endpoint [POST]: /register 
Checks if email doesn't exist in database
Hashes password
Creates User(200)
Returns Error' response if:
Email already exists(409),
Password hashing failed(500), 
Saving user in db failed(500),

### CODE REVIEW

File: src/server.ts [hunk 1]
```diff
@@ -21,11 +21,13 @@ mongoose
 
 // bring in routes
 import productRoutes from './routes/product';
+import authRoutes from './routes/auth';
 
 //middleware
```
Comment (thatsddr) [inline src/server.ts:26]: I don't know for sure but I think that the frontend won't be able to make requests unless it is authorized by the CORS middleware

**Review by thatsddr — APPROVED** [review]
The code looks good, the only doubt I have is that the frontend will not be authorized to make requests to the backend

**Comment by thatsddr** [comment]
Screen Shot 2021-03-16 at 12 27 37
I've done some testing and CORS is a problem even after proxying the backend. What should we do?

File: src/server.ts (hunk 1 above)
Comment (Darjusch) [reply src/server.ts:26]: Okay good point i will implement a cors middleware.

Comment (Darjusch) [reply src/server.ts:26]: I will create a new pull request for that

Comment (Darjusch) [reply src/server.ts:26]: can you create an issue for that with your screenshot?

**Comment by d-pettersson** [comment]
Works for me - not sure how to test "failed hash" or "saving user in db"

**Review by 5h3rr1ll — APPROVED** [review]
Now it should be fine.

**Pull request merged**